pkg/live/load.go

```diff
@@ -58,26 +67,52 @@ func (e *MultipleInventoryInfoError) Error() string {
 // for inventory information inside Kptfile resources.
 // It returns the resources in unstructured format and the inventory information.
 // If no inventory information is found, that is not considered an error here.
-func Load(f util.Factory, path string, stdIn io.Reader) ([]*unstructured.Unstructured, kptfilev1.Inventory, error) {
+func Load(f util.Factory, path, rgfile string, stdIn io.Reader) ([]*unstructured.Unstructured, kptfilev1.Inventory, error) {
 	if path == "-" {
-		return loadFromStream(f, stdIn)
+		return loadFromStream(f, stdIn, rgfile)
 	}
-	return loadFromDisk(f, path)
+	return loadFromDisk(f, path, rgfile)
 }
 
 // loadFromStream reads resources from the provided reader and returns the
 // filtered resources and any inventory information found in Kptfile resources.
 // If there is more than one Kptfile in the stream with inventory information, that
 // is considered an error.
-func loadFromStream(f util.Factory, r io.Reader) ([]*unstructured.Unstructured, kptfilev1.Inventory, error) {
+func loadFromStream(f util.Factory, r io.Reader, rgfile string) ([]*unstructured.Unstructured, kptfilev1.Inventory, error) {
 	var stdInBuf bytes.Buffer
 	tee := io.TeeReader(r, &stdInBuf)
 
+	// Check if stream contains inventory info.
 	invInfo, err := readInvInfoFromStream(tee)
 	if err != nil {
 		return nil, kptfilev1.Inventory{}, err
 	}
 
+	// Check resourcegroup file for inventory information if file is specified.
+	if rgfile != "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, kptfilev1.Inventory{}, err
+		}
+
+		diskInv, err := readInvInfoFromDisk(cwd, rgfile)
+		if err != nil {
+			return nil, kptfilev1.Inventory{}, err
+		}
+
+		if diskInv.Name != "" && invInfo.Name != "" {
```

Comment:
Can we check this against empty structs instead of only checking the `Name` property?

Reply:
Have updated the code to check if the inventory is valid or not, rather than only using the `Name` property. Currently this check enforces `Name` and `Namespace` to not be empty. Also added new test cases to check for this!